### PR TITLE
[WIP] Admin Redesign

### PR DESCRIPTION
**Pull request by coderberry**
The following pages need to be moved into the new UIKIT design:

### Admin
- [ ] Dashboard

### Users
- [ ] Index
- [ ] New
- [ ] Edit

### Teams (admin only for now)
- [ ] Index
- [ ] New
- [ ] Edit

### Common
- [ ] User Profile > Show
- [ ] User Profile > Edit
- [ ] Change Password
- [ ] Referrals

### Publisher
- [ ] Publisher Dashboard

### Campaigns
- [ ] Index
- [ ] New

### Campaign
- [ ] Overview
- [ ] Edit
- [ ] Changelog

### Campaign Analyze
- [ ] Daily
- [ ] Property
- [ ] Country
- [ ] Creative

### Assets
- [ ] Index
- [ ] Upload

### Creatives
- [ ] Index
- [ ] Overview
- [ ] New
- [ ] Edit

### Properties
- [ ] Index
- [ ] New

### Property
- [ ] Overview
- [ ] Edit
- [ ] Changelog

### CODE REVIEW

File: app/controllers/campaigns_controller.rb [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
     else
-      campaigns = campaigns.where(user: current_user)
+      @campaigns = campaigns.where(user: current_user)
     end
-    campaigns = @campaign_search.apply(campaigns)
-    @pagy, @campaigns = pagy(campaigns)
+    @campaigns = @campaigns.includes(user: :organization)
 
-    render "/campaigns/for_user/index" if @user
+    respond_with @campaigns
```
Comment (hopsoft) [inline app/controllers/campaigns_controller.rb]: Pretty sure we can omit `respond_with` as this mimics default Rails behavior.

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -243,4 +243,11 @@ def sparkline(values, width: 100, height: 30, stroke_width: 3, color: "green", f
     svg_class = "sparkline sparkline--#{color}#{" sparkline--filled" if filled}"
     render "/@shared/widgets/sparkline_graph", values: values, width: width, height: height, stroke_width: stroke_width, svg_class: svg_class
   end
+
+  # TODO: This is pretty hacky
+  def subnav_li(title, icon, link)
+    active_link_to link, wrap_tag: :li, class_active: 'uk-active' do
+      "<span class=\"uk-visible@s\">#{title}</span><span class=\"uk-hidden@s\" uk-icon=\"icon: #{icon}\"></span>".html_safe
+    end
+  end
```
Comment (hopsoft) [inline app/helpers/application_helper.rb]: May be worth moving to an `@shared` partial.

File: app/models/concerns/campaigns/presentable.rb [hunk 3]
```diff
@@ -20,6 +20,10 @@ def duration
       return nil if total_operative_days.zero?
       distance_of_time_in_words(start_date.beginning_of_day, end_date.end_of_day, scope: "datetime.distance_in_words.short")
     end
+    
+    def remaining_budget
+       total_budget.to_f - total_consumed_budget.to_f
+    end
```
Comment (hopsoft) [inline app/models/concerns/campaigns/presentable.rb]: This logic already exists under `Campaigns::Budgetable#total_remaining_budget` https://github.com/gitcoinco/code_fund_ads/blob/master/app/models/concerns/campaigns/budgetable.rb#L17-L21

File: app/views/layouts/admin.html.erb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  <body class="">
+    <%= render "/@shared/admin/navbar" %>
+    <%= render "/@shared/admin/breadcrumbs" %>
+    <%= render "/@shared/admin/subnav" %>
+
+    <%= yield %>
+
+    <%= render "/@shared/admin/offcanvas" %>
+  </body>
+</html>
```
Comment (hopsoft) [inline app/views/layouts/admin.html.erb]: `admin` is a bit of a misnomer here, but I suppose it works.

File: app/controllers/campaigns_controller.rb (hunk 1 above)
Comment (coderberry) [reply app/controllers/campaigns_controller.rb]: I had no idea. That's awesome that they did that.

**Comment by hopsoft** [comment]
Closing this in favor of the new UI design based on Bootstrap.

**Pull request closed without merging**